.github/workflows/reusable-test.yml

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+name: CI
+
+on:
+  workflow_call:
+    inputs:
+      os-variant:
+        default: 'ubuntu-20.04'
+        type: string
+      build-config:
+        default: 'dev-env'
+        type: string
+      python-version:
+        default: '3.9'
+        type: string
+      coverage-report:
+        default: false
+        type: boolean
+
+jobs:
+  test:
+    name: tests ${{ inputs.os-variant }}, py${{ inputs.python-version }}
+    runs-on: ${{ inputs.os-variant }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Setup conda environment
+        uses: mamba-org/provision-with-micromamba@main
```

Comment:
Using conda/mamba because...? Do we depend on anything that is conda-only?

Reply:
It was just much faster to build the virtual environment with mamba.
Should this be avoided...?

Reply:
I am asking you you are not using `pip` (with `tox` to handle the virtual envs), e.g., like in scippnexus or plopp.

Reply:
Using mamba&tox was sliiightly faster but basically no difference,
and as you said, it'll be better to use ``ci.txt`` instead of ``lime-dev.yml``, so I updated it to use pip&tox.

### Using mamba&tox
![image](https://user-images.githubusercontent.com/17974113/233070532-d9cec50e-80ff-4cf5-a1f8-48d4488a498f.png)
[link to the action summary](https://github.com/scipp/beamlime/actions/runs/4743039071)

### Using pip&tox
![image](https://user-images.githubusercontent.com/17974113/233070380-4bba7184-583c-4066-9cc9-eb54db899577.png)
[link to the action summary](https://github.com/scipp/beamlime/actions/runs/4733193251)